src/Uno.UI/UI/Xaml/Controls/MenuFlyout/CascadingMenuHelper.cs

```diff
@@ -0,0 +1,976 @@
+
+		// PointerEntered event handler that shows the sub menu
+		// whenever the pointer is over the sub menu owner.
+		// In case of touch, the sub menu item will be shown by
+		// PointerReleased event.
+		internal void OnPointerEntered(PointerRoutedEventArgs args)
+		{
+			bool handled = false;
+
+			m_isPointerOver = true;
+
+			handled = (args.Handled);
+
+#if CMH_DEBUG
+    (DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnPointerEntered - handled=%d.", this, handled));
+#endif // CMH_DEBUG
+
+			if (!handled)
+			{
+				var owner = m_wpOwner?.Target as ISubMenuOwner;
+
+#if CMH_DEBUG
+        (DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnPointerEntered - owner=0x%p.", this, owner));
+#endif // CMH_DEBUG
+
+				if (owner != null)
+				{
+					ISubMenuOwner parentOwner;
+					parentOwner = (owner.ParentOwner);
+
+#if CMH_DEBUG
+            (DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnPointerEntered - parentOwner=0x%p.", this, parentOwner));
+#endif // CMH_DEBUG
+
+					if (parentOwner != null)
+					{
+						parentOwner.CancelCloseSubMenu();
+					}
+				}
+
+				Pointer pointer;
+				var pointerDeviceType = PointerDeviceType.Touch;
+
+				pointer = args.Pointer;
+				pointerDeviceType = pointer.PointerDeviceType;
+
+#if CMH_DEBUG
+				(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnPointerEntered - pointerDeviceType=%d.", this, pointerDeviceType));
+#endif // CMH_DEBUG
+
+				if (pointerDeviceType != PointerDeviceType.Touch)
+				{
+					CancelCloseSubMenu();
+
+					EnsureDelayOpenMenuTimer();
+
+#if CMH_DEBUG
+            (DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnPointerEntered - Starting m_delayOpenMenuTimer.", this));
+#endif // CMH_DEBUG
+
+					m_delayOpenMenuTimer.Start();
+
+					UpdateOwnerVisualState();
+				}
+
+				args.Handled = true;
+			}
+
+
+		}
+
+		// PointerExited event handler that ensures we close the sub menu
+		// whenever the pointer leaves the current sub menu or
+		// the main presenter. If the exited point is on the sub menu owner
+		// or the sub menu, we want to keep the sub menu open.
+		internal void OnPointerExited(
+			PointerRoutedEventArgs args,
+		   bool parentIsSubMenu)
+		{
+			bool handled = false;
+
+			m_isPointerOver = false;
+			m_isPressed = false;
+
+			handled = args.Handled;
+
+#if CMH_DEBUG
+    (DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnPointerExited - handled=%d, parentIsSubMenu=%d.", this, handled, parentIsSubMenu));
+#endif // CMH_DEBUG
+
+			if (m_delayOpenMenuTimer != null)
+			{
+#if CMH_DEBUG
+        (DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnPointerExited - Stopping m_delayOpenMenuTimer.", this));
+#endif // CMH_DEBUG
+
+				m_delayOpenMenuTimer.Stop();
+			}
+
+			var ownerAsUIE = m_wpOwner?.Target as FrameworkElement;
+
+			if (!handled && ownerAsUIE != null && ownerAsUIE.IsLoaded)
+			{
+				Pointer pointer;
+				var pointerDeviceType = PointerDeviceType.Touch;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'pointerDeviceType'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)